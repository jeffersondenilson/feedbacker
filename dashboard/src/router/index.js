import { createRouter, createWebHistory } from 'vue-router'

// route level code-splitting
// this generates a separate chunk (about.[hash].js) for this route
// which is lazy-loaded when the route is visited.

const Home = () => import('../views/Home/index')
const Feedbacks = () => import('../views/Feedbacks/index')
const Credentials = () => import('../views/Credentials/index')

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/credentials',
    name: 'Credentials',
    component: Credentials,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  // process.env.BASE_URL
  history: createWebHistory('/'),
  routes
})

export default router
