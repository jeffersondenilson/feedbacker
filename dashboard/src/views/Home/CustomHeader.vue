<template>
  <div>custom header</div>
</template>

<script>
export default {
  name: 'CustomHeader'
}
</script>
