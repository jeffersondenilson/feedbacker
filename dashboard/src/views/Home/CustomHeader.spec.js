import { createRouter, createWebHistory } from 'vue-router'
import { shallowMount } from '@vue/test-utils'
import CustomHeader from './CustomHeader'
import { routes } from '@/router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<CustomHeader />', () => {
  it('should render custom-header correctly', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(CustomHeader, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit create-account from top button', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(CustomHeader, {
      global: {
        plugins: [router]
      }
    })

    const buttonTop = wrapper.find('[data-test=create-account-button-top]')
    await buttonTop.trigger('click')

    expect(wrapper.emitted()).toHaveProperty('create-account')
  })

  it('should emit create-account from middle button', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(CustomHeader, {
      global: {
        plugins: [router]
      }
    })

    const buttonMiddle = wrapper
      .find('[data-test=create-account-button-middle]')
    await buttonMiddle.trigger('click')

    expect(wrapper.emitted()).toHaveProperty('create-account')
  })

  it('should emit login from button', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(CustomHeader, {
      global: {
        plugins: [router]
      }
    })

    const loginButton = wrapper.find('[data-test=login-button]')
    await loginButton.trigger('click')

    expect(wrapper.emitted()).toHaveProperty('login')
  })
})
