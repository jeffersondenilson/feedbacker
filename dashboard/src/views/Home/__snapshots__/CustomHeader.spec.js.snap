// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomHeader /> should render custom-header correctly 1`] = `
<header class="header">
  <div class="header-group">
    <div class="flex items-center justify-between py-10">
      <div class="w-28 lg:w-36"><img class="w-full" src="" alt="Logo"></div>
      <div class="flex text-sm sm:text-base"><button class="px-3 sm:px-6 py-2 font-bold rounded-full text-white focus:outline-none" data-test="create-account-button-top"> Crie uma conta </button><button class="px-4 sm:px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none" data-test="login-button"> Entrar </button></div>
    </div>
    <div class="flex flex-col md:mt-28">
      <h1 class="text-4xl font-black text-white"> Tenha um feedback. <br> E faça seus clientes mais <br class="hidden lg:inline-block"> felizes! </h1>
      <p class="text-lg font-medium text-white"> Receba ideias, reclamações e feedbacks com um <br class="hidden lg:inline-block"> simples widget na página. </p>
      <div><button class="my-8 px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none" data-test="create-account-button-middle"> Crie uma conta grátis </button></div>
    </div>
  </div>
</header>
`;
